types/wlr_output.c

```diff
@@ -252,6 +252,27 @@ void wlr_output_make_current(struct wlr_output *output) {
 	output->impl->make_current(output);
 }
 
+static void output_fullscreen_surface_render(struct wlr_output *output,
+		struct wlr_surface *surface, const struct timespec *when) {
```

Comment:
Is the whole point of handling the rendering here to support direct scanout later?

Reply:
Yes. This is a software fallback in case overlay planes and direct scan-out isn't supported.